x-pack/solutions/security/plugins/security_solution/public/data_view_picker/components/data_view_picker/index.tsx

```diff
@@ -0,0 +1,131 @@
+import { useDispatch, useSelector } from 'react-redux';
+
+import { DataView, type DataViewListItem } from '@kbn/data-views-plugin/public';
+import type { DataViewPickerScopeName } from '../../constants';
+import { useKibana } from '../../../common/lib/kibana';
+import { DEFAULT_SECURITY_SOLUTION_DATA_VIEW_ID } from '../../constants';
+import { useDataView } from '../../hooks/use_data_view';
+import { sharedStateSelector } from '../../redux/selectors';
+import { shared } from '../../redux/slices';
+import { useSelectDataView } from '../../hooks/use_select_data_view';
+
+export const DataViewPicker = memo((props: { scope: DataViewPickerScopeName }) => {
+  const dispatch = useDispatch();
+  const selectDataView = useSelectDataView();
+
+  const {
+    services: { dataViewEditor, data, dataViewFieldEditor, fieldFormats },
+  } = useKibana();
+
+  const closeDataViewEditor = useRef<() => void | undefined>();
+  const closeFieldEditor = useRef<() => void | undefined>();
+
+  // TODO: this should be disabled for the default data views probably, eg. `security-solution-default`
+  // const canEditDataView =
+  // Boolean(dataViewEditor?.userPermissions.editDataView()) || !dataView.isPersisted();
+  const canEditDataView = true;
+
+  const { dataView } = useDataView(props.scope);
+
+  const dataViewId = dataView?.id;
+
+  const createNewDataView = useCallback(() => {
+    closeDataViewEditor.current = dataViewEditor.openEditor({
+      onSave: async (newDataView) => {
+        dispatch(shared.actions.addDataView(newDataView));
+        selectDataView({ id: newDataView.id, scope: [props.scope] });
+      },
+      allowAdHocDataView: true,
+    });
+  }, [dataViewEditor, dispatch, props.scope, selectDataView]);
+
+  const onFieldEdited = useCallback(() => {}, []);
+
+  const editField = useMemo(() => {
+    if (!canEditDataView) {
+      return;
+    }
+    return async (fieldName?: string, _uiAction: 'edit' | 'add' = 'edit') => {
+      if (!dataViewId) {
+        return;
+      }
+
+      const dataViewInstance = await data.dataViews.get(dataViewId);
+      closeFieldEditor.current = await dataViewFieldEditor.openEditor({
+        ctx: {
+          dataView: dataViewInstance,
+        },
+        fieldName,
+        onSave: async () => {
+          onFieldEdited();
+        },
+      });
+    };
+  }, [canEditDataView, dataViewId, data.dataViews, dataViewFieldEditor, onFieldEdited]);
+
+  const addField = useMemo(
+    () => (canEditDataView && editField ? () => editField(undefined, 'add') : undefined),
+    [editField, canEditDataView]
+  );
+
+  const handleChangeDataView = useCallback(
+    (id: string) => {
+      selectDataView({ id, scope: [props.scope] });
+    },
+    [props.scope, selectDataView]
+  );
+
+  const handleEditDataView = useCallback(() => {}, []);
+
+  const triggerConfig = useMemo(() => {
+    if (dataView.id === DEFAULT_SECURITY_SOLUTION_DATA_VIEW_ID) {
+      return {
+        label: 'Default Security Data View',
```

Comment:
I figure the plan here is to configure this on the actual `dataView` in the future?

Reply:
probably, right now the name is not available I think